File: java/bitcoin/Number/NumberFactory2.java
import java.math.BigInteger;
import java.util.Random;

public class NumberFactory2 extends NumberFactory
{
  @Override public Number zero(){ return Number2.ZERO; }
  @Override public Number one(){ return Number2.ONE; }
  @Override public Number fromBytes(int signum, byte[] val){
    return new Number2(new BigInteger(signum, val));}
  @Override public Number random(int numBits, Random rnd){
    return new Number2(new BigInteger(numBits, rnd));}
  @Override public Number fromBigInteger(BigInteger n){
    return new Number2(n);}


}

File: c#/bitcoin/Number/NumberFactory2.cs
using System;
using System.Numerics;
using System.Security.Cryptography;
using Random = System.Security.Cryptography.RandomNumberGenerator;

public class NumberFactory2 : NumberFactory
{
  public override Number Zero(){ return Number2.ZERO; }
  public override Number One(){ return Number2.ONE; }
  public override Number FromBytes(int sign, byte[] val){
    return Number2.FromBytes(sign, val); }
  public override Number Random(int numBits, Random rnd){
    return Number2.Random(numBits, rnd); }
  public override Number FromBigInteger(BigInteger n){
    return new Number2(n);  }
}
